searx/doc.py

```diff
@@ -0,0 +1,199 @@
+                self._raw_content = f.read()
+        return self._raw_content
+
+    @property
+    def content(self):
+        """Content of the page (rendered in a Jinja conntext)"""
+        if self._content is None:
+            ctx = self.get_ctx()
+            template = jinja2.Environment().from_string(self.raw_content)
+            self._content = template.render(**ctx)
+        return self._content
+
+    @property
+    def title(self):
+        """Title of the content (without any markup)"""
+        if self._title is None:
+            for l in self.raw_content.split('\n'):
+                if l.startswith('# '):
+                    self._title = l.strip('# ')
+                    break
+        return self._title
+
+    def get_ctx(self):  # pylint: disable=no-self-use
+        """Jinja context to render :py:obj:`DocPage.content`"""
+
+        def _md_link(name, url):
+            url = url_for(url)
+            if self.base_url:
+                url = self.base_url + url
+            return "[%s](%s)" % (name, url)
+
+        def _md_search(query):
+            url = '%s?q=%s' % (url_for('search'), urllib.parse.quote(query))
+            if self.base_url:
+                url = self.base_url + url
+            return '[%s](%s)' % (query, url)
+
+        ctx = {}
+        ctx['GIT_URL'] = GIT_URL
+        ctx['get_setting'] = get_setting
+        ctx['link'] = _md_link
+        ctx['search'] = _md_search
+
+        return ctx
+
+    def render(self):
+        """Render / return content"""
+        return self.content
+
+
+class MistletoePage(DocPage):
+    """A HTML page of the :py:obj:`online documentation <OnlineDoc>`."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._html = None
+
+    @property
+    def html(self):
+        """HTML representation of this page"""
+        return self.render()
+
+    def render(self):
+        """Render Markdown (CommonMark_) to HTML by using mistletoe_.
+
+        .. _CommonMark: https://commonmark.org/
+        .. _mistletoe: https://github.com/miyuchina/mistletoe
+
+        """
+        if self._html is None:
+            self._html = mistletoe.markdown(self.content)
+        return self._html
+
+
+_DOC_FOLDER = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'doc')
+
+
+class OnlineDoc:  # pylint: disable=too-few-public-methods
+    """Cached rendering of the online documentation a SearXNG instance has.
+
+    :param page_class: render online documentation by :py:obj:`DocPage` parser.
+    :type page_class: :py:obj:`DocPage`
+    """
+
+    def __init__(self, page_class: typing.Type[DocPage], base_url=None):
+        self.page_class = page_class
+        self.base_url = base_url
+        self.CACHE: typing.Dict[tuple, DocPage] = {}
+
+        # future: could be set from settings.xml
+
+        self.folder: str = _DOC_FOLDER
+        """location of the Markdwon files"""
+
+        self.i18n_origin: str = 'en'
```

Comment:
Because this is the origin document, the other languages are just translations.

Reply:
weblate uses ["source language"](https://docs.weblate.org/en/weblate-4.11/admin/projects.html#component-source-language).

Since we use locale everywhere (except for the query language), I suggest `self.locale_source`